### DISCUSSION TITLE
Next13 app - Navigation slow on Vercel

### DISCUSSION THREAD

**Discussion by EmanuelU** [vercel/community#1109] [Help] [answered]
Hi,

I have a very small app that I am building. 
When I run npm run build && npm run start locally, navigation is instant and the app is very bare bones.
When I deploy my app to Vercel, each load (for each navigation) is about 2 seconds long. 

What could be causing this? 
I am using the new app directory

![image](https://user-images.githubusercontent.com/24460536/205335696-13f5c72f-e92c-4a1e-8cc4-7ec33f70a498.png)

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @EmanuelU! Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can see? There are too many possible causes to be able to pick the correct one without getting more information. A minimal reproduction is usually the fastest way for us to see what is going wrong. 

You should also reach out to the [Next.js community](https://github.com/vercel/next.js/discussions), if you have not already, in case someone there already has a solution.

**Reply by EmanuelU** [reply]
Hi there and thanks for replying.
As I was producing the minimal example, I figured out the culprit, but not the solution (or why it works so fast on local machine vs Vercel hosting).

I had a client component using "useSearchParams" to get some data from the URL, apparently that is slowing down everything.

I need to know current page/url somehow, in a client component. The component itself is used in the Layout file

**Reply by amyegan** [reply]
Great find Emanuel! I see https://github.com/vercel/next.js/pull/43603 was opened to fix a `useSearchParams` bug. That should speed up navigation for you once it is merged and released. Thank you for reporting and sharing what you found!

**Reply by EmanuelU** [reply]
I restructured my app to use paths instead, for now :) Thanks
